src/librustdoc/html/render/mod.rs

```diff
@@ -884,9 +884,11 @@ fn render_assoc_item(
                     ItemType::TyMethod
                 };
 
-                href(did.expect_real(), cx)
-                    .map(|p| format!("{}#{}.{}", p.0, ty, name))
-                    .unwrap_or_else(|| format!("#{}.{}", ty, name))
+                match href(did.expect_real(), cx) {
+                    Ok(p) => Some(format!("{}#{}.{}", p.0, ty, name)),
+                    Err(HrefError::UnknownLocation) => None,
+                    Err(_) => Some(format!("#{}.{}", ty, name)),
+                }
```

Comment:
This does not look like the right fix. I expect if you change `cargo doc --no-deps` to `cargo doc` you'll still run into the same issue (you can test with `cargo deadlinks --dir target/doc`).

The proper fix is probably to use `#tymethod` for external methods and `#method` for local ones? *Something* about the if on line 881 looks wrong, I'm not sure exactly what.

Reply:
I tried building docs with and without dependencies for `serde-encrypt`, no deadlinks on `vzip`.

Yes, `#method` is for the link on the local method, and is available on the mouse over anchor. `#tymethod` would be the link to the trait definition, on the function name, but it's not available here and this fix removes it.